lib/parser/file-parser.ts

```diff
@@ -20,12 +20,14 @@ export class FileParser extends ParserBase {
     super(new ParserState(new ParserPosition(), filePath));
     this.originalText = text;
     this.text = text;
-    
-    this.file = new OFile(this.text, this.state.fileUri, this.originalText, this.lexerTokens);
-    const lexer = new Lexer(this.originalText, this.file);
-    this.lexerTokens.push(...lexer.lex(this.file));
-    this.file.lexerTokens = this.lexerTokens;
-
+    // Because of cat eats tail and typescript reasons lexer Token array has to be predefined.
+    // Lexer takes the array and returns reference to the same array
+    // In the end lexerTokens === this.file.lexerTokens === this.lexerTokens
+    const lexerTokens: [] = [];
```

Comment:
```suggestion
    const lexerTokens: OLexerTokens[] = [];
```

Reply:
This is on purpose. The lexer is expecting an empty array, otherwise funny stuff might happen.